src/lib.rs

```diff
@@ -2574,6 +2582,37 @@ impl<'a> TryFrom<&'a Path> for &'a Utf8Path {
     }
 }
 
+/// Converts an [`OsStr`] to a [`Utf8Path`].
+///
+/// Returns the original [`OsStr`] if it is not valid UTF-8.
```

Comment:
This should return a `FromOsStrError` similar to `FromPathError`. (And the same for `Utf8PathBuf` above.)